core/embed/boardloader/main.c

```diff
@@ -280,6 +285,23 @@ int main(void) {
 #if defined USE_SD_CARD
   sdcard_init();
 
+#ifdef STM32U5
+  secret_ensure_initialized();
```

Comment:
This chunk is probably incorrectly nested. You initialize the secret only in case both `USE_SD_CARD` and `STM32U5` are true. This will not be the case on T3B1.

Reply:
moved https://github.com/trezor/trezor-firmware/commit/c8d2a20f2e4b9e9415b74a69eb4ae0d9c22ec207